### PR TITLE
Added Tier as option to aws_ssm_parameter_store module

### PR DESCRIPTION
**Pull request by 116davinder**
##### SUMMARY
Added Tier as parameter for parameter store.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
community.aws.aws_ssm_parameter_store

##### ADDITIONAL INFORMATION
Original Pull Request: ansible/ansible#61933
Original Issue: ansible/ansible#59738

if this doesn't get merged then I have published same module in personal repository: https://github.com/116davinder/ansible.missing_collection

### CODE REVIEW

**Comment by 116davinder** [comment]
@ansibullbot ready_for_review

**Comment by 116davinder** [comment]
Original Pull Request: https://github.com/ansible/ansible/pull/61933
Original Issue: https://github.com/ansible/ansible/issues/59738

File: plugins/modules/aws_ssm_parameter_store.py [hunk 1]
```diff
@@ -65,10 +65,14 @@
     choices: ['never', 'changed', 'always']
     default: changed
     type: str
-author:
```
Comment (markuman) [inline plugins/modules/aws_ssm_parameter_store.py:68]: Why removing the other authors?

**Review by markuman — CHANGES_REQUESTED** [review]
You also need a changelog fragment and you can mentioned when the `tier` parameter was added. 1.3.0 will be the next release imho.

File: plugins/modules/aws_ssm_parameter_store.py (hunk 1 above)
Comment (116davinder) [reply plugins/modules/aws_ssm_parameter_store.py:68]: 02:06 ERROR: Found 1 validate-modules issue(s) which need to be resolved:
02:06 ERROR: plugins/modules/aws_ssm_parameter_store.py:0:0: invalid-documentation: DOCUMENTATION.author: Invalid author for dictionary value @ data['author']. Got ['Davinder Pal ( @116davinder )', 'Nathan Webster (@nathanwebsterdotme)', 'Bill Wang (@ozbillwang) <ozbillwang@gmail.com>', 'Michael De La Rue (@mikedlr)'] (75%)

ref: https://app.shippable.com/github/ansible-collections/community.aws/runs/1040/1/console

**Comment by 116davinder** [comment]
> You also need a changelog fragment and you can mentioned when the `tier` parameter was added. 1.3.0 will be the next release imho.

@markuman, any example or reference for it?

**Comment by tremble** [comment]
https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to

File: plugins/modules/aws_ssm_parameter_store.py (hunk 1 above)
Comment (markuman) [reply plugins/modules/aws_ssm_parameter_store.py:68]: Just grep in plugins/modules `grep -rn -A 4  author *`.  
Maybe the missing indent of the elements is wrong?

Comment (116davinder) [reply plugins/modules/aws_ssm_parameter_store.py:68]: I will give one more try for it.

**Comment by 116davinder** [comment]
@tremble / @markuman , when you have time can you review it?

**Comment by 116davinder** [comment]
@ansibullbot ready_for_review

**Comment by aston-r** [comment]
@markuman Could you please review the changes since I am also interested in this functionality?

File: plugins/modules/aws_ssm_parameter_store.py [hunk 2]
```diff
@@ -65,10 +65,20 @@
     choices: ['never', 'changed', 'always']
     default: changed
     type: str
+  tier:
+    description:
+      - Paramter Store Tier Type
+    required: false
+    choices: ['Standard', 'Advanced', 'Intelligent-Tiering']
+    default: Standard
+    type: str
```
Comment (markuman) [inline plugins/modules/aws_ssm_parameter_store.py:74]:
```suggestion
    type: str
    version_added: 1.5.0
```

**Review by markuman — CHANGES_REQUESTED** [review]

**Review by markuman — APPROVED** [review]
LGTM

**Comment by 116davinder** [comment]
@markuman , when it will be merged?
I checked the sanity, it failed with some unknown error any idea? how to resolve it?
ref: https://dashboard.zuul.ansible.com/t/ansible/build/3f0e49fc280748d2b8ace828e4bc91b9

**Comment by markuman** [comment]
recheck

File: plugins/modules/aws_ssm_parameter_store.py [hunk 3]
```diff
@@ -65,10 +65,21 @@
     choices: ['never', 'changed', 'always']
     default: changed
     type: str
+  tier:
+    description:
+      - Paramter Store Tier Type
```
Comment (tremble) [inline plugins/modules/aws_ssm_parameter_store.py:70]:
```suggestion
      - Parameter store tier type.
```

**Review by tremble — APPROVED** [review]
Couple of minor niggles that I'll merge.  As long as the tests pass I'll get this merged.

**Comment by tremble** [comment]
Hi @116davinder,

Sorry it's taken so long to get this merged.  Thanks for taking the time to raise the PR and add this functionality.

**Pull request merged**